### DISCUSSION TITLE
Add cascaded method invocation feature

### DISCUSSION THREAD

**Discussion by xshaheen** [dotnet/csharplang#4484] [General]
## Summary

Add method cascades. Method cascades provide a syntactic sugar for situations where the object of a method invocation might otherwise have to be repeated.

## Motivation

Make it easer to write fluent API.
With cascades, we will gain fluent API on the .NET base library.

## Syntax

The use of `..` is to cascaded method invocation operation. The `..` syntax invokes a method (field or property) but discards the result, and returns the original object instead.

```cs
List<int> list = new();

list
..Add(1)
..Add(2)
..Add(3);

String s = (new StringBuilder()
..Append("Hello")
..Append(" ")
..Append("Word")
).ToString();
```

**Comment by HaloFour** [comment]
https://github.com/dotnet/csharplang/discussions/781

**Reply by Kimi-Arthur** [reply]
This is quite different from what's proposed in #781 , which messes with typing and require void for the original method return type.

**Reply by HaloFour** [reply]
Those would be design and implementation details for effectively the same feature request.

**Reply by Kimi-Arthur** [reply]
[MgSam](https://github.com/MgSam)'s objection won't apply for the design in this proposal. Syntax is key here as it's for language design.

**Reply by HaloFour** [reply]
IMO that still doesn't necessitate a completely separate discussion.  The idea is the same.
